### PR TITLE
Travis cache virtualenv

### PR DESCRIPTION
**Pull request by dannyroberts**
I noticed before when working on this that even just downloading the `commcarehq-venv` repo takes a while, so this could theoretically speed things up significantly

### CODE REVIEW

**Comment by snopoke** [comment]
:+1:

**Comment by snopoke** [comment]
Killed build for first commit

**Comment by dannyroberts** [comment]
I'm going to restart the previous build, see comment on the second: "this is a separate commit so the previous one can build the cache
and so we can easily measure the difference in speed"

**Comment by snopoke** [comment]
👍 sorry

On 12 October 2015 at 13:16, Daniel Roberts <notifications@github.com>
wrote:

> I'm going to restart the previous build, see comment on the second: "this
> is a separate commit so the previous one can build the cache
> and so we can easily measure the difference in speed"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8590#issuecomment-147368154>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Comment by dannyroberts** [comment]
@snopoke no worries, wasn't obvious from the PR

**Comment by dannyroberts** [comment]
Looks like this approach doesn't quite work. I'll re-PR just the pip-caching part of it

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 language: python
 sudo: false
-cache: pip
+cache:
+  directories:
+    - $HOME/.cache/pip
+    - $HOME/virtualenv
```
Comment (benrudolph) [inline .travis.yml]: did you check out this guy's answer? http://stackoverflow.com/a/19460794/835696

Comment (benrudolph) [inline .travis.yml]: specifically the bit about `download-cache`

Comment (dannyroberts) [inline .travis.yml]: I think I don't understand the difference between caching `.cache/pip` and using a download cache. If we cache both when only one is necessary then that's more stuff to copy on startup, so it's unclear to me what the tradeoff is. What's your take on it? For now, I think https://github.com/dimagi/commcare-hq/pull/8608 is good start?

Comment (snopoke) [inline .travis.yml]: the `--download-cache` options is also deprecated from pip>=6.0

On 13 October 2015 at 07:04, Daniel Roberts <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/dimagi/commcare-hq/pull/8590#discussion_r41827267>:
>
> > @@ -1,6 +1,9 @@
> >  language: python
> >  sudo: false
> > -cache: pip
> > +cache:
> > +  directories:
> > +    - $HOME/.cache/pip
> > +    - $HOME/virtualenv
>
> I think I don't understand the difference between caching .cache/pip and
> using a download cache. If we cache both when only one is necessary then
> that's more stuff to copy on startup, so it's unclear to me what the
> tradeoff is. What's your take on it? For now, I think #8608
> <https://github.com/dimagi/commcare-hq/pull/8608> is good start?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8590/files#r41827267>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

Comment (benrudolph) [inline .travis.yml]: sorry you're totally right. i think i lost my train of thought when researching that yesterday

**Pull request closed without merging**